components/job-orchestration/job_orchestration/retention/streams_handler.py

```
@@ -0,0 +1,93 @@
+    recovery_file = logs_directory / f"{STREAMS_RETENTION_HANDLER_NAME}.tmp"
+    targets_buffer = TargetsBuffer(recovery_file)
+
+    with pymongo.MongoClient(results_cache_config.get_uri()) as results_cache_client:
+        results_cache_db = results_cache_client.get_default_database()
+        stream_collection = results_cache_db.get_collection(
+            results_cache_config.stream_collection_name
+        )
+        # Find documents where _id (and thus creation time) is earlier
+        retention_filter = {MONGODB_ID_KEY: {"$lt": expiry_oid}}
+        results = stream_collection.find(retention_filter)
+        object_ids_to_delete: List[ObjectId] = list()
```

Pylint (3.3.7)

[refactor] Consider using [] instead of list()

(R1734)